lib/publiccloud/instance.pm

```diff
@@ -210,22 +210,32 @@ sub wait_for_ssh
     $args{proceed_on_failure} //= 0;
     $args{username}           //= $self->username();
     my $start_time = time();
+    my $check_port = 1;
 
-    # Check port 22
+    # Looping until reaching timeout or passing two conditions :
+    # - SSH port 22 is reachable
+    # - journalctl got message about reaching one of certain targets
     while ((my $duration = time() - $start_time) < $args{timeout}) {
-        last if (script_run('nc -vz -w 1 ' . $self->{public_ip} . ' 22', quiet => 1) == 0);
+        if ($check_port && script_run('nc -vz -w 1 ' . $self->{public_ip} . ' 22', quiet => 1) == 0) {
```

Comment:
IMHO there is a logic error. If `$check_port = 1` but the `nc` command fails, it would set `$check_port = 0` and immediately switch over to the `journalctl` branch. I think what you want to do is something like the following:
```suggestion
        if ($check_port) {
            $check_port = 0 if (script_run('nc -vz -w 1 ' . $self->{public_ip} . ' 22', quiet => 1) == 0);
        } elsif ($self->run_ssh_command(cmd => ...
```